confluent_kafka/schema_registry/avro.py

```diff
@@ -0,0 +1,330 @@
+
+        if not isinstance(self._auto_register, bool):
+            raise ValueError("auto.register.schemas must be a boolean value")
+
+        subject_name_func = conf_copy.pop('subject.name.strategy')
+
+        if not callable(subject_name_func):
+            raise ValueError("subject.name.strategy must be callable.")
+
+        self._subject_name_func = subject_name_func
+
+        if len(conf_copy) > 0:
+            raise ValueError("Unrecognized property(ies) {}"
+                             .format(conf_copy.keys()))
+
+        # convert schema_str to Schema instance
+        schema = _schema_loads(schema_str)
+        schema_dict = loads(schema.schema_str)
+        parsed_schema = parse_schema(schema_dict)
+        #  https://github.com/fastavro/fastavro/issues/415
+        schema_name = parsed_schema.get('name', schema_dict['type'])
+
+        self._schema = schema
+        self._schema_name = schema_name
+        self._parsed_schema = parsed_schema
+        self._hash = hash(schema.schema_str)
+
+    def __hash__(self):
+        return self._hash
+
+    def __call__(self, obj, ctx):
+        """
+        Serializes an object to the Confluent Schema Registry binary format
+        for Avro.
+
+        Arguments:
+            obj (object): object instance to encode
+            ctx (SerializationContext): Metadata pertaining to the serialization
+                operation.
+
+        Note:
+            None objects are represented as Kafka Null.
+
+        Raises:
+            SerializerError if any error occurs encoding obj
+
+        Returns:
+            bytes: Confluent Schema Registry formatted Avro bytes
+
+        """
+        if obj is None:
+            return None
+
+        subject = self._subject_name_func(self._schema_name, ctx)
+
+        if self._auto_register and subject not in self._known_subjects:
+            self._schema_id = self._registry.register_schema(subject,
+                                                             self._schema)
+            self._known_subjects.add(subject)
+        elif not self._auto_register and subject not in self._known_subjects:
+            registered_schema = self._registry.lookup_schema(subject,
+                                                             self._schema)
+            self._schema_id = registered_schema.schema_id
+            self._known_subjects.add(subject)
+
+        if self.to_dict is not None:
+            obj = self.to_dict(obj, ctx)
+
+        with _ContextStringIO() as fo:
+            # Write the magic byte and schema ID in network byte order (big endian)
+            fo.write(pack('>bI', _MAGIC_BYTE, self._schema_id))
+            # write the record to the rest of the buffer
+            schemaless_writer(fo, self._parsed_schema, obj)
+
+            return fo.getvalue()
+
+
+class AvroDeserializer(Deserializer):
+    """
+    AvroDeserializer decodes bytes written in the Schema Registry
+    format for Avro to an object.
```

Comment:
It's the Schema Registry Avro format, rather than to or from Avro. Since we much with the encoding its not really Avro but I was also avoiding making it sound like we forked avro